js/main.js

```diff
@@ -0,0 +1,146 @@
+const searchButton = document.querySelector('.search-panel button');
+const resultsContainer = document.querySelector('.resultsContainer');
+
+function fetchLocationName() {
+  const inputField = document.querySelector('.search-panel input');
+  const query = inputField.value.trim();
+
+  if (!query) {
+    alert("Please enter a search term.");
+    return;
+  }
+
+  document.querySelector('.container.homepage').style.display = 'none';
+
+  resultsContainer.style.display = 'block';
+
+  const searchAPI = `https://api.content.tripadvisor.com/api/v1/location/search?key=5859D966AFCD4FBB9F8883B3C5C5951C&searchQuery=${encodeURIComponent(query)}&language=en`;
+
+  fetch(searchAPI)
+    .then(response => {
+      if (!response.ok) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
+      }
+      return response.json();
+    })
+    .then(data => {
+      const locationID = data.data[0].location_id;
+      console.log("The locationID is:", locationID);
+      if (locationID) {
+        fetchLocationData(locationID);
+      } else {
+        console.error('No location found for the search query.');
+      }
+    })
+    .catch(error => {
+      console.error('Error fetching location ID: ', error);
+    });
+}
+
+// Function to fetch location details using the location ID
+function fetchLocationData(locationID) {
+  const locationDetailsAPI = `https://api.content.tripadvisor.com/api/v1/location/${locationID}/details?key=5859D966AFCD4FBB9F8883B3C5C5951C&language=en&currency=USD`;
+  const locationPhotosAPI = `https://api.content.tripadvisor.com/api/v1/location/${locationID}/photos?key=5859D966AFCD4FBB9F8883B3C5C5951C&language=en`;
+
+  Promise.all([
+    fetch(locationDetailsAPI).then(response => response.json()),
+    fetch(locationPhotosAPI).then(response => response.json())
+  ])
+  .then(([detailsData, photosData]) => {
+    renderLocation(detailsData);
+    if (photosData.data) {
+      renderPhotos(photosData.data);
+    }
+  })
+  .catch(error => {
+    console.error('Error fetching data:', error);
+  });
+}
+
+
+function renderLocation(data) {
+  resultsContainer.innerHTML = '';
+
+  const {name, description, web_url} = data;
+
+  const locationName = document.createElement('h3');
+  locationName.textContent = name;
+
+  const locationDescription = document.createElement('p');
+  locationDescription.textContent = description;
+
+  const locationLink = document.createElement('a');
+  locationLink.href = web_url;
+  locationLink.textContent = 'Read more on TripAdvisor';
+  locationLink.target = '_blank';
+
+  resultsContainer.appendChild(locationName);
+  resultsContainer.appendChild(locationDescription);
+  resultsContainer.appendChild(locationLink);
```

Comment:
**hint:** you can use [this method](https://developer.mozilla.org/en-US/docs/Web/API/Element/append) to make this more DRY 😉